### DISCUSSION TITLE
libmetal on Linux without kernel modules (uio_pdrv_genirq etc.)?

### DISCUSSION THREAD

**Discussion by infn-ke** [OpenAMP/libmetal#150] [Q&A]
Is it supported to run libmetal on Linux without having the kernel modules uio_pdrv_genirq, uio_dmem_genirq, vfio-pci and uio_pci_generic? The code tries to do `modprobe` on these in `metal_linux_probe_driver`. I want to use openamp on Linux but without being tied to those kernel modules.

**Comment by edmooring** [comment]
Unfortunately, there is no way to do that right now. De-coupling the libmetal device abstraction on linux from UIO would be challenging.

**Comment by infn-ke** [comment]
I tried to decouple it by replacing the `metal_sys_init` with an implementation that simply registers the generic bus `metal_generic_bus` rather than doing modprobe on all the kernel modules. Then I just register a generic device describing a shared memory (`metal_register_generic_device`). Wouldn't that be a viable path?

**Comment by edmooring** [comment]
Sorry about the delay in responding. That might work. I may need something similar, so I will probably try something like it.
